src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey <T>(items:T[], key: keyof T): GroupsMap<T> {
+  const arr:GroupsMap<T> = {};
```

Comment:
'arr' is a terrible name for this object

Reply:
agree